dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,285 @@
+    
+    final static int maxRecusionDepth = 10;
+    
+    public String averageSizePretty(final Map<String, Object> cacheMap) {
+
+        return UtilMethods.prettyByteify(averageSize(cacheMap));
+
+    }
+
+    /**
+     * Takes a map and pulls random values from it to calculate an average object size
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long averageSize(final Map<String, Object> cacheMap) {
+
+        if (cacheMap.size() == 0) {
+            return 0;
+        }
+
+        long totalSize = 0;
+
+        final int numberToSample = Math.min(cacheMap.size(), standardSampleSize);
+
+
+        for (int i = 0; i < numberToSample; i++) {
+            List<String> keysAsArray = new ArrayList<String>(cacheMap.keySet());
+            Random r = new Random();
+            Object object = cacheMap.get(keysAsArray.get(r.nextInt(keysAsArray.size())));
+            if (object instanceof Optional && ((Optional) object).isPresent()) {
+                object = ((Optional) object).get();
+            }
+            if (object instanceof Optional && !((Optional) object).isPresent()) {
+                return 0;
+            }
+            totalSize += sizeOf(object);
+        }
+        return totalSize / numberToSample;
+
+    }
+
+    /**
+     * Tries to use the Unsafe class to calculate an objects size. If this does not work, it will fall
+     * back on the serialzied size (which less than accurate).
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long sizeOf(Object object) {
+        long retainedSize = 0;
+        try {
+            retainedSize = retainedSize(object);
+        } catch (Throwable t) {
+            Logger.infoEvery(this.getClass(),
+                            "Unable to use the Unsafe.class to size cache objects, falling back to serialization:"
+                                            + t.getMessage(),
+                            60000);
+        }
+
+        return retainedSize > 0 ? retainedSize : sizeOfSerialized(object);
+
+    }
+
+
+    final ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+
+    /**
+     * Serializes an object to attempt to find an object's size
+     * 
+     * @param object
+     * @return
+     */
+    long sizeOfSerialized(Object object) {
+
+        if (object == null) {
+            return 0;
+        }
+
+        if (!(object instanceof Serializable)) {
+            Logger.warn(this.getClass(), object.getClass() + " not serializable: " + object);
+            return -1;
+        }
+
+        if (object instanceof String) {
+            return ((String) object).length() * 8;
+
+        }
+
+        byteOutputStream.reset();
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteOutputStream)) {
+            objectOutputStream.writeObject(object);
+            objectOutputStream.close();
+
+            return byteOutputStream.toByteArray().length;
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)